ui/app/src/components/CreateFolderDialog/CreateFolderContainer.tsx

```diff
@@ -68,16 +73,44 @@ export function CreateFolderContainer(props: CreateFolderContainerProps) {
 		? name !== value && (itemExists || lookupItemByPath(newFolderPath, itemLookupTable) !== UNDEFINED)
 		: itemExists || lookupItemByPath(newFolderPath, itemLookupTable) !== UNDEFINED;
 	const [moveFolderAck, setMoveFolderAck] = useState(false);
-	const isValid = !isBlank(name) && !folderExists && (!rename || (name !== value && moveFolderAck));
+	const [fetchingAffectedPackages, setFetchingAffectedPackages] = useState(false);
+	const [packagesInWorkflow, setPackagesInWorkflow] = useState(undefined);
+	const containsItemsInWorkflow = packagesInWorkflow?.length > 0;
+	const [cancelPackagesAck, setCancelPackagesAck] = useState(false);
+	const isValid =
+		!isBlank(name) &&
+		!folderExists &&
+		(!rename || (name !== value && moveFolderAck)) &&
+		(!rename || !containsItemsInWorkflow || (containsItemsInWorkflow && cancelPackagesAck));
 
 	useEffect(() => {
 		if (item && rename === false) {
 			setSelectedItem(item);
 		}
 	}, [item, rename]);
 
+	useEffect(() => {
+		if (rename) {
+			// If renaming, check if the folder contains items in workflow
+			setFetchingAffectedPackages(true);
+			fetchAffectedPackages(site, path, true).subscribe({
+				next: (packages) => {
+					setFetchingAffectedPackages(false);
+					setPackagesInWorkflow(packages);
+				},
+				error: () => {
+					setFetchingAffectedPackages(false);
+				}
+			});
+		} else {
+			setPackagesInWorkflow(undefined);
+		}
```

Comment:
Let's make the else into a guard statement.
```suggestion
		if (!rename) {
		setPackagesInWorkflow(undefined);
		return;
		}
			// If renaming, check if the folder contains items in workflow
			setFetchingAffectedPackages(true);
			fetchAffectedPackages(site, path, true).subscribe({
				next: (packages) => {
					setFetchingAffectedPackages(false);
					setPackagesInWorkflow(packages);
				},
				error: () => {
					setFetchingAffectedPackages(false);
				}
			});
		
```